packages/api/models/collections.js

```diff
@@ -52,6 +53,23 @@ const validateCollection = (collection) => {
 
   // Check that each file.regex matches against file.sampleFileName
   collection.files.forEach((file) => checkRegex(file.regex, file.sampleFileName));
+
```

Comment:
We already validate each `file.regex` and `checksumFor` has to be identical to one of those values, otherwise we throw an error, so it's being validated that way. Did you have something else in mind?

Reply:
As long as that's always going to be the case then I guess it's fine. I'd feel a little more comfortable if we were actually validating that this is a good regex, though.

Reply:
I guess I'm not 100% on what that means. Pass it into a `RegExp` constructor?

Reply:
I guess that, as long as we're verifying that `checksumFor` is identical to some `regex`, and we're also verifying that each `regex` matches a `sampleFileName`, then an invalid `checksumFor` regex wouldn't slip through.